### PR TITLE
Preview Large Datasets

### LINKED ISSUE #22: Processing node for preview datasets
We need new processing node to define one more processing flow (preview datasets). For that, we need to implement a class inheriting from `BaseProcessingNode`, and add that class name to the `ORDERED_NODE_CLASSES array`. new processing node will iterate over all resources and add processors that are generating first 10000 rows and views to the ones that have type `derived/json` 

## Acceptance criteria

* [ ] after able to run `load_previews` and `load_views` from #23 

## Tasks 

* [x] define new class for preview processing node `class DerivedPreviewProcessingNode` that will inherit `DerivedFormatProcessingNode`
* [x] in the `DerivedPreviewProcessingNode` override `get_artifacts` and Iterate over existing artefacts (resources) 
* [x] check for the ones that are derived/json (or csv?) 
* [x] and pass the processors that we will need to get the desired resources and views.
* [x] add `DerivedPreviewProcessingNode` to the `ORDERED_NODE_CLASSES` in correct position (last in the list)

## Analysis 

Example how refactored `basic_nodes.py` may look like 

```
class DerivedFormatProcessingNode(BaseProcessingNode):

  def get_artifacts(self):
    for artifact in self.available_artifacts:
      if artifact.datahub_type == 'derived/csv':
        datahub_type = 'derived/preview' 
        resource_name = artifact.resource_name + '_preview.'
        output = ProcessingArtifact(
        datahub_type, resource_name,
        [artifact], [],
        [(new processors + old goes here)],
                True
            )
            yield output

class DerivedPreviewProcessingNode(...):
  def __init__(self):
    ... # (all same)
```

### PR DESCRIPTION
**Pull request by zelima**
Processing preview resources

* two new processors for getting first N rows and updating descriptor with views
* load modified resources handles with views
* Bug fix re dependency names (see commit message)
* New processing node for previews dependent on `derived/csv`

closes #22 #23 
refs #16

### CODE REVIEW

File: datapackage_pipelines_assembler/nodes/basic_nodes.py [hunk 1]
```diff
@@ -56,3 +56,42 @@ def get_artifacts(self):
...
+
+class DerivedPreviewProcessingNode(BaseProcessingNode):
+    def __init__(self, available_artifacts, outputs):
+        super(DerivedPreviewProcessingNode, self).__init__(available_artifacts, outputs)
+        self.fmt = 'json'
+
+    def get_artifacts(self):
+        for artifact in self.available_artifacts:
+            if artifact.datahub_type == 'derived/csv':
+                datahub_type = 'derifdfdved/preview'
```
Comment (anuveyatsu) [inline datapackage_pipelines_assembler/nodes/basic_nodes.py]: it seems like a typo: 'derifdfdved/preview'

File: datapackage_pipelines_assembler/nodes/basic_nodes.py [hunk 2]
```diff
@@ -56,3 +56,42 @@ def get_artifacts(self):
...
+                resource_name = artifact.resource_name + '_preview'
```
Comment (anuveyatsu) [inline datapackage_pipelines_assembler/nodes/basic_nodes.py]: so is it derived csv resource name + '_preview'?

File: datapackage_pipelines_assembler/nodes/basic_nodes.py [hunk 3]
```diff
@@ -56,3 +56,42 @@ def get_artifacts(self):
...
+                      {
+                          'name': artifact.resource_name,
+                          'update': {
+                              'name': resource_name,
+                              'format': self.fmt,
+                              'path': 'data/{}.{}'.format(resource_name, self.fmt),
+                              'datahub': {
+                                'type': "derived/preview",
+                                'derivedFrom': [
+                                    artifact.resource_name
```
Comment (anuveyatsu) [inline datapackage_pipelines_assembler/nodes/basic_nodes.py]: again is this original resource name?

File: datapackage_pipelines_assembler/processors/load_preview.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from datapackage_pipelines.wrapper import spew, ingest
+
+parameters, datapackage, res_iter = ingest()
+
+
+def generate_priview(res):
```
Comment (anuveyatsu) [inline datapackage_pipelines_assembler/processors/load_preview.py]: a typo: 'priview' => 'preview'

File: datapackage_pipelines_assembler/processors/load_preview.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    for i, row in enumerate(res):
+        if i < int(parameters.get('limit', 10000)):
+            yield row
+
+
+def process_resources(res_iter_):
+    for res in res_iter_:
+        yield generate_priview(res)
```
Comment (anuveyatsu) [inline datapackage_pipelines_assembler/processors/load_preview.py]: a typo: 'priview' => 'preview'

File: datapackage_pipelines_assembler/nodes/basic_nodes.py (hunk 2 above)
Comment (zelima) [reply datapackage_pipelines_assembler/nodes/basic_nodes.py]: yes, so if resource name is `vix-daily` preview name would be `vix-daily_csv_priview`  (as vix-daily_csv is the name of the resource it is iterating over)

File: datapackage_pipelines_assembler/nodes/basic_nodes.py (hunk 3 above)
Comment (zelima) [reply datapackage_pipelines_assembler/nodes/basic_nodes.py]: No this is the name of the resource that it is derived. Eg: vix-daily_csv

Comment (zelima) [reply datapackage_pipelines_assembler/nodes/basic_nodes.py]: No this is name of derived resource in this case `resource-name_csv`

File: datapackage_pipelines_assembler/nodes/basic_nodes.py [hunk 6]
```diff
@@ -56,3 +56,42 @@ def get_artifacts(self):
             False
         )
         yield output
+
```
Comment (akariv) [inline datapackage_pipelines_assembler/nodes/basic_nodes.py]: let's move this to a 'preview_node.py' file.

**Pull request merged**